storage/store.go

```diff
@@ -144,6 +144,39 @@ func verifyKeys(start, end proto.Key, checkEndKey bool) error {
 	return nil
 }
 
+// verifyTransactionKeys verifies that requests that update transaction
+// keys have valid request keys matching with transaction keys.
+//
+// Request keys need to be same as transaction keys to guarantee that
+// at most one update command runs for a given transaction. (The command
+// queue ensures that.)
+func verifyTransactionKeys(args proto.Request) error {
+	header := args.Header()
+	if !proto.IsTransactionKeyUpdate(args) {
+		return nil
+	}
+	var txn *proto.Transaction
+	switch tArgs := args.(type) {
```

Comment:
Hmm, we can, but we need to check only `txn != nil` as keys in `RequestHeader` are transaction keys. I could be convinced either way, but I thought that adding too many checks in `executeCmd` might be overkill.

Yes, we can remove the checking in `EndTransaction`, but I wasn't sure either. It might be still nice to have the checks inside each command.